### PR TITLE
[Feature #159965314] List users functionality

### PR DESCRIPTION
**Pull request by mikenthiwa**
#### What does this PR do?
Allows authenticated users to list profiles of all authors.

#### Description of Task to be completed?
A user is able to view all profiles of all authors and is not able to edit the profiles profiles

#### How should this be manually tested?
$ git clone https://github.com/andela/ah-magnificent6.git
$ git checkout ft-list-user-functionality-159965314
$ cd ah-magnificent6
$ cp .env-sample .env [and set the environment variables]
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip install -r requirements.txt
(venv) $ python manage.py migrate
(venv) $ python manage.py runserve

Using [Postman](https://www.getpostman.com/) navigate to 
User registration: `POST {your-server-root}:8000/api/user/signup`
```
{
    "email": "user1@user.user",
    "username": "user1",
    "password": "User$#@!"
}
```
User Login: `POST {your-server-root}:8000/api/user/login`
```
{
    "email": "user1@user.user",
    "password": "User$#@!"
}
```

Use the token from the response in the login as the authentication header while listing all authors

List all authors
In Postman client: `GET {your-server-root}:8000/api/profiles/authors`
screen shot 2018-09-18 at 11 52 16

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2192364/stories/159965314

### CODE REVIEW

**Comment by meshack-mbuvi** [comment]
I have noted that our coverage is complaining. Please fix that as we agreed to keep it up. Meanwhile, I will be going through your code.

**Comment by mikenthiwa** [comment]
@meshack-mbuvi 
Test coverage has been improved.
Kindly continue going through the PR.

**Comment by meshack-mbuvi** [comment]
It looks good to me unless have missed something.

**Comment by pngg** [comment]
looks good to me too :)

File: authors/settings.py [hunk 1]
```diff
@@ -184,7 +184,7 @@
 
     # Add pagination default class and number of objects in page
     'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
-    'PAGE_SIZE': 20
+    'PAGE_SIZE': 10
```
Comment (parseen254) [inline authors/settings.py]: @mikenthiwa why was this reduced?

**Review by parseen254 — CHANGES_REQUESTED** [review]
Also, squash your migrations. Recommended resource: https://stackoverflow.com/questions/40028586/how-to-squash-recent-django-migrations

File: authors/settings.py (hunk 1 above)
Comment (mikenthiwa) [reply authors/settings.py]: I was testing my feature using a page size of 10.
I will revert it to 20 since it works fine

Comment (mikenthiwa) [reply authors/settings.py]: @parseendavid Returned page size to 20
Squashed migrations

**Pull request merged**